scipy/stats/_stats_py.py

```diff
@@ -6987,35 +6996,49 @@ def ttest_1samp(a, popmean, axis=0, nan_policy='propagate',
     953
 
     """
-    a, axis = _chk_asarray(a, axis)
+    xp = array_namespace(a)
+    a, axis = _chk_asarray(a, axis, xp=xp)
 
     n = a.shape[axis]
     df = n - 1
 
-    mean = np.mean(a, axis)
+    mean = xp.mean(a, axis=axis)
     try:
-        popmean = np.squeeze(popmean, axis=axis)
+        popmean = xp.asarray(popmean)
+        popmean = xp.squeeze(popmean, axis=axis) if popmean.ndim > 0 else popmean
     except ValueError as e:
         raise ValueError("`popmean.shape[axis]` must equal 1.") from e
     d = mean - popmean
-    v = _var(a, axis, ddof=1)
-    denom = np.sqrt(v / n)
+    v = _var(a, axis=axis, ddof=1)
+    denom = xp.sqrt(v / n)
 
     with np.errstate(divide='ignore', invalid='ignore'):
-        t = np.divide(d, denom)[()]
-    prob = _get_pvalue(t, distributions.t(df), alternative)
+        t = xp.divide(d, denom)
+        t = t[()] if t.ndim == 0 else t
```

Comment:
perhaps worth adding a helper if this is going to keep coming up?

Reply:
It will keep coming up. I'm not sure why, but I'm conflicted about a helper for now. Maybe because it's part of a bigger issue that I think it's worth reminding us about when we read the code? NumPy reducing functions return a different type of object (scalar instead of array) depending on input dimensionality. If NumPy were to switch to returning 0d arrays after reducing operations on 1d elements, we could follow suit and get rid of `[()]` everywhere.